pkg/syncer/controllermanager/controllermanager.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllermanager
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/kcp-dev/kcp/pkg/logging"
+)
+
+const (
+	ControllerNamePrefix = "syncer-controller-manager-"
+)
+
+type InformerSource struct {
+	Subscribe func(id string) <-chan struct{}
```

Comment:
what is id?

Reply:
This is not "subscribe". It is "Events()". Subscribe would return and keep you subscribed.

Reply:
Actually, this is "subscribe". With subscribe I would expect either an event handler or the channel passed in, not returned (and logic to remove it when closed).

Reply:
This is the same as the DDSIF Subscribe() method.
```
// Subscribe registers for informer/discovery change notifications, returning a channel to which change notifications
// are sent.
func (d *GenericDiscoveringDynamicSharedInformerFactory[Informer, Lister, GenericInformer]) Subscribe(id string) <-chan struct{} {
```
`id` is the identifier of the subscriber, since there might be several subscribers subscribing to receive events from the `InformerSource`.